model/task/task.go

```diff
@@ -1273,6 +1260,35 @@ func SetTasksScheduledTime(tasks []Task, scheduledTime time.Time) error {
 	return nil
 }
 
+// GetTaskIdBetweenIds gets the task between two task given that they are
+// from the same project, requester, build variant, and display name. The
+// order of the ID's does not matter and if the task passed cannot have a
+// middle (i.e. it is sequential tasks or the same task) it will return the
+// the first task given.
+func GetMidwayTask(t1, t2 Task) (*Task, error) {
+	// The tasks should be the same build variant, display name, project, and requester.
+	if t1.BuildVariant != t2.BuildVariant {
```

Comment:
Small suggestion to use a Grip error catcher to accumulate all the errors and return them, rather than return on first error (so that the caller gets all the errors, not just the first error).

```go
catcher := grip.NewBasicCatcher() // Makes an error accumulator
catcher.ErrorfWhen(t1.BuildVariant != t2.BuildVariant, "given tasks have differing build variants '%s' and '%s', t1.BuildVariant, t2.BuildVariant`) // Conditionally adds formatted error
catcher.ErrorfWhen(...)
...
if catcher.HasErrors() {
    return nil, catcher.Resolve()
}

// Continue now that we know input is valid
```